extensions/notebook/src/book/bookModel.ts

```diff
@@ -0,0 +1,221 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as yaml from 'js-yaml';
+import * as glob from 'fast-glob';
+import { BookTreeItem, BookTreeItemType } from './bookTreeItem';
+import { maxBookSearchDepth, notebookConfigKey } from '../common/constants';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import * as nls from 'vscode-nls';
+import { IJupyterBookToc, IJupyterBookSection } from '../contracts/content';
+
+const localize = nls.loadMessageBundle();
+//const exists = promisify(fs.exists);
+
+export class BookModel implements azdata.nb.NavigationProvider {
+	public BookPath: string;
+	private _bookItems: BookTreeItem[] = [];
+	private _allNotebooks = new Map<string, BookTreeItem>();
+	public openAsUntitled: boolean = false;
+	private _extensionContext: vscode.ExtensionContext;
+	private _tableOfContentPaths: string[] = [];
+	readonly providerId: string = 'BookNavigator';
+	// For testing
+	private _errorMessage: string;
```

Comment:
What's this used for?

Reply:
I copied over the code whatever Lucy had and made changes on top of it. This is something she had in her original implementation. Looks like created a errorMessage property but isn't using it anywhere. Not sure why did it. Let me clean it up.